### ISSUE TITLE
Support Multi Pub Key

### ISSUE THREAD

**Issue by SatoshiNakamotoBitcoins** [JeanLucPons/Kangaroo#1]
Jean-Luc,

At first clap your hands for our French Master!

Thank you for this great app.

An question...how does the multi pub key search work...does it searches sequentially (start searching for the first pubkey in the in.txt file and move on) or searches it for ALL the pub keys at the same time and quotes a private key when it finds one of the pubkeys in the in.txt file.

Like to learn from you!

Greetings Satoshi

**Comment by dem10** [comment]
There was an error starting
D:\BSGS\Kang>Kangaroo in.txt
Kangaroo v1.0
Error: in.txt not enough arguments

**Comment by JeanLucPons** [comment]
Yes for the moment pubKey are searched sequentially.
Could you show you in.txt ?
I'm currently on the GPU support

**Comment by SatoshiNakamotoBitcoins** [comment]
Thank you Jean-Luc for your answer...I tested something with your pubkeys in the readme as EXAMPLE of Usage...

**Comment by JeanLucPons** [comment]
The method used here use distinguished point. This is a very efficient method for detecting collisions (no need of a Bloom filter).
For multiple key a solution would be the set wild kangaroos to several pub key but it will decrease the probability to find a match. To be investigated...

**Comment by dem10** [comment]
dear Jean. And could you make a kangaroo or BSGS for opencl? I understand that you are not the master of this business, but there is no one to do this except you. But the rest just throw empty words and you do your job.

**Comment by hamnaz** [comment]
multiple pubkeys  REMINDER :)

**Comment by GGG9** [comment]
Not moving to the next pubkeys until it finds the previous 
even if it comes to the end of key space it continues to search.
checked with short key space when the first pubkey not in key space it search indefinitely and does not go to the second pubkey.

**Comment by JeanLucPons** [comment]
Yes, will add a stop option to give up the search if a threshold according to the expected number of operation is reached.

**Comment by virus-cyber** [comment]
JeanLucPons no 
Multi Pub Key
search

**Comment by kpot87** [comment]
@JeanLucPons how do you think when multi pub key search will be realized ?

**Comment by SatoshiNakamotoBitcoins** [comment]
Guys be patient... @JeanLucPons will deliver...like always...;-)

**Comment by JeanLucPons** [comment]
Yes, I added load/save/work
The multi key will be done with the help of theses new features.
The idea is to allow to pre compute large tame kangaroo file and to solve multiple key using this file.
Then when a key is solved, all the wild will become tame and more keys are solved more chance to solve the others....

**Comment by SatoshiNakamotoBitcoins** [comment]
@JeanLucPons We all appreciat your hard work and thanks for adding the multi key support!!

Can you give an example for properly use the added flags (load/save/work) in order to execute search with a list containing multi pub keys.

Thnx in advance.

Cheers Satoshi

**Comment by JeanLucPons** [comment]
Yes,
I will add some note about this on the readme. It is a bit tricky.
Multi key support is not yet supported, for this you will need first to create a large tame array for a given range and then attack keys with it.

**Comment by GGG9** [comment]
@JeanLucPons 
Hi 
How to use maxStep and how to calculate maxStep can be an example?

**Comment by GGG9** [comment]
I loaded several Pub Keys into the program, but after Continue the work from "-i save.work", the program writes "Keys: 1"

**Comment by kpot87** [comment]
@JeanLucPons hi any news of realize multipubkeysearch? Tnx

**Comment by roi35** [comment]
Hello @JeanLucPons any news multi pub key release ?

**Comment by h2o10** [comment]
simultaneous pubkey are not possible with pollard rho kangaroo.

**Comment by Shadow145-cpu** [comment]
> I loaded several Pub Keys into the program, but after Continue the work from "-i save.work", the program writes "Keys: 1"

I have the same problem

**Comment by SatoshiNakamotoBitcoins** [comment]
@Shadow145-cpu That is correct it will only load the first pubkey of the original in.txt file which created the save.work file...(multi pub key is not supported YET...)

**Comment by Shadow145-cpu** [comment]
So we can wait for the multi pub :)

**Comment by SatoshiNakamotoBitcoins** [comment]
@Shadow145-cpu I hope so...;-)

**Comment by kpot87** [comment]
@SatoshiNakamotoBitcoins i don’t think so
